hello-world-attest-fn-call/main.go

```diff
@@ -0,0 +1,18 @@
+package main
```

Comment:
Looking at the docs, the functions, some names used in the example, and where certain pieces of code are located, there may be a slight disconnect with the model for running functions. Let's circle up to try and clean that up.

Reply:
take result pattern out to an example on error handling